Marlin/src/lcd/menu/menu_ubl.cpp

```diff
@@ -315,37 +333,39 @@ void _lcd_ubl_invalidate() {
 void _lcd_ubl_build_mesh() {
   START_MENU();
   BACK_ITEM(MSG_UBL_TOOLS);
-  #if HAS_HEATED_BED
-    GCODES_ITEM(MSG_UBL_BUILD_MESH_M1, PSTR(
-      "G28\n"
-      "M190 S" STRINGIFY(PREHEAT_1_TEMP_BED) "\n"
-      "M109 S" STRINGIFY(PREHEAT_1_TEMP_HOTEND) "\n"
-      "G29 P1\n"
-      "M104 S0\n"
-      "M140 S0"
-    ));
-    GCODES_ITEM(MSG_UBL_BUILD_MESH_M2, PSTR(
-      "G28\n"
-      "M190 S" STRINGIFY(PREHEAT_2_TEMP_BED) "\n"
-      "M109 S" STRINGIFY(PREHEAT_2_TEMP_HOTEND) "\n"
-      "G29 P1\n"
-      "M104 S0\n"
-      "M140 S0"
-    ));
-  #else
-    GCODES_ITEM(MSG_UBL_BUILD_MESH_M1, PSTR(
-      "G28\n"
-      "M109 S" STRINGIFY(PREHEAT_1_TEMP_HOTEND) "\n"
-      "G29 P1\n"
-      "M104 S0"
-    ));
-    GCODES_ITEM(MSG_UBL_BUILD_MESH_M2, PSTR(
-      "G28\n"
-      "M109 S" STRINGIFY(PREHEAT_2_TEMP_HOTEND) "\n"
-      "G29 P1\n"
-      "M104 S0"
-    ));
-  #endif
+  #if PREHEAT_COUNT
+    #if HAS_HEATED_BED
+      #define BUILD_MESH_GCODE_ITEM(p)  GCODES_ITEM(MSG_UBL_BUILD_MESH_M##p, PSTR(  \
+                                          "G28\n" \
+                                          "M190 S" STRINGIFY(PREHEAT_##p##_TEMP_BED) "\n" \
+                                          "M109 S" STRINGIFY(PREHEAT_##p##_TEMP_HOTEND) "\n"  \
+                                          "G29 P1\n"  \
+                                          "M104 S0\n" \
+                                          "M140 S0" \
+                                        ))
+    #else
+      #define BUILD_MESH_GCODE_ITEM(p)  GCODES_ITEM(MSG_UBL_BUILD_MESH_M##p, PSTR(  \
+                                          "G28\n" \
+                                           "M109 S" STRINGIFY(PREHEAT_##p##_TEMP_HOTEND) "\n" \
```

Comment:
All these preheat items are editable. One of the maintainers of UBL needs to catch up on that.